Push instance-to-node mapping to cw log, configure new log group

Jennifer603 opened the pull request:
### Description of changes
* Push instance-to-node mapping to cw log
* configured new log group nodemap

### Tests
* Manually tested by creating cluster and check log event nodemap

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Commit:
Push instance-to-node mapping to cw log, configure new log group

davprat commented on cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py:
```diff
@@ -45,15 +56,19 @@ def gethostname():
     """Add a "log_group_name": log_group_name to every config."""
     for config in configs:
+        log_group_name = default_log_group_name
+        group_key = config.get("log_group_key")
```
I think you need to provide a default here: `.get("log_group_key", None)` otherwise this call will throw an error if "log_group_key" is not present in `config`.

davprat commented on cookbooks/aws-parallelcluster-environment/files/cloudwatch/write_cloudwatch_agent_json.py:
```diff
@@ -45,15 +56,19 @@ def gethostname():
+        log_group_name = default_log_group_name
+        group_key = config.get("log_group_key")
+        if group_key:
+            log_group_name = log_group_name_map.get(group_key)
```
Same here: better to provide a default value rather than raise an exception.

davprat reviewed (changes requested):
Looks good - I left a couple comments to address, plus we need to figure out why the checks are failing and fix those issues as well.

Commit:
Export default ENV when PATH is required for command execution

exportfs and authconfig execution was failing with:
No such file or directory - exportfs/authconfig

* `default_env true` documentation: https://docs.chef.io/resources/execute/#properties
* Related issue: https://github.com/sous-chefs/nfs/issues/106

Signed-off-by: Enrico Usai <usai@amazon.com>

davprat reviewed (approved):
lgtm
